solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
I think we still want to keep the solver to default at the construction, it would set the solver to studentSnopt in your case. Then when adding constraints and decision variables, if the solver is studentSnopt and the constraint/variable size exceeds the limit, we can either switch to other solvers by calling setSolver('default'), or throw an error message.